### PR TITLE
Preserves x_forwarded_host on https

### LINKED ISSUE #141: FORCE_HTTPS doesn't persist requested name.
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?
```
{
   "name": "",
   "build": "",
   "support": "https://support.pivotal.io",
   "version": 0,
   "description": "",
   "authorization_endpoint": "https://login.g4.app.cloud.comcast.net",
   "token_endpoint": "https://uaa.g4.app.cloud.comcast.net",
   "min_cli_version": "6.23.0",
   "min_recommended_cli_version": "6.23.0",
   "api_version": "2.94.0",
   "app_ssh_endpoint": "ssh.g4.app.cloud.comcast.net:2222",
   "app_ssh_host_key_fingerprint": "12:c6:ed:80:48:83:30:28:b5:f3:4a:c0:13:f5:c1:19",
   "app_ssh_oauth_client": "ssh-proxy",
   "doppler_logging_endpoint": "wss://doppler.g4.app.cloud.comcast.net:443",
   "routing_endpoint": "https://api.g4.app.cloud.comcast.net/routing"
}
cf version 6.35.0+5c7de3e32.2018-03-08
```

What version of the buildpack you are using?
`staticfile_buildpack-cached-v1.4.21.zip`

If you were attempting to accomplish a task, what was it you were attempting to do?
Force HTTPS behind a CNAME/load balancer

What did you expect to happen?
The app should redirect to https on the requested name:
`http://cname.comcast.com --> 301 --> https://cname.comcast.com`

What was the actual behavior?
The app redirected to the local instance fqdn: 
`http://cname.comcast.com --> 301 --> https://myapp.g1.app.cloud.comcast.net`

Please confirm where necessary:
- [ ] I have included a log output
- [ ] My log includes an error message
- [x] I have included steps for reproduction

### PR DESCRIPTION
**Pull request by dmarmugi**
Closes #141. Updates .ForceHTTPS block to preserve the
x-forwarded-host on https redirects.

Thanks for contributing to the buildpack. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:
Updates `src/finalize/data.go` .ForceHTTPS block to preserve `X-Forwarded-Host` on https redirect.

* An explanation of the use cases your change solves
If an app is behind a CNAME, the https redirect should preserve that CNAME. 

* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `develop` branch

* [x] I have added an integration test

I wasn't sure whether to add logic to the existing force_https integration test or create a new one.

### CODE REVIEW

**Comment by sclevine** [comment]
Sorry for the delay! See #137. We've decided to respect the various `X-Forwarded-*` headers in the buildpack, so we can definitely accept this PR and add the other headers as well.

Are you able to add tests for this change?

**Comment by dmarmugi** [comment]
Glad to hear it. I'll work on adding tests, though I invite extra scrutiny as this will be my first time working in Go. 

Regarding #137 - is this PR intended to address that issue as well? That is, do we want a test for that scenario as well?

**Comment by dmarmugi** [comment]
I've added one test to `finalize/finalize_test.go`. It feels like the test at `integration/deploy_https_redirect_test.go` should also be updated, but I'm not seeing where a `cutlass.App` preserves or cares about the `x-forwarded-*` headers in order to set it up. 

I'm having trouble installing bosh-lite to actually run the tests, or I'd be able to hack away at this more.

**Comment by dmarmugi** [comment]
* Got local cf-deployment working and got tests running. 
* Realized how to pass in the header to Cutlass app. 
* Test added on feature branch. Verified it failed on `master`, so seems like a valid and sufficient test to me.

**Comment by ameyer-pivotal** [comment]
Thanks for the update @dmarmugi. We'll take a look!

File: src/staticfile/finalize/data.go [hunk 1]
```diff
@@ -80,15 +80,25 @@ http {
...
    {{if .ForceHTTPS}}
+	 set $updated_host $host;
+	 if ($http_x_forwarded_host != "") {
+       set $updated_host $http_x_forwarded_host;
+     } 
+
      if ($http_x_forwarded_proto != "https") {
-       return 301 https://$host$request_uri;
-     }
+	   return 301 https://$updated_host$request_uri;
```
Comment (ameyer-pivotal) [inline src/staticfile/finalize/data.go]: @dmarmugi This is minor, but would you mind fixing some of the formatting in this file?

Comment (dmarmugi) [reply src/staticfile/finalize/data.go]: sure @ameyer-pivotal, i think the ide put in tabs. two options, though -- the `{{if .ForceHTTPS}}` block _was_ on a 3-space indent. I could move the whole block to fall in line with the rest of the file, or just make sure that my update falls in with the formatting that was there before.

Comment (tylerphelan) [reply src/staticfile/finalize/data.go]: We just want it to be consistent, so either option is fine with us 😉

**Comment by dmarmugi** [comment]
Hi, Any word on when this will get merged in? I've got several projects pointing at a fork that I'd prefer to point against the main track. 
Thanks,
-david

**Comment by jfeeny** [comment]
@dmarmugi sorry for the delay!

@sclevine would you like to prioritize this?

**Comment by sclevine** [comment]
Bumped the priority of the PR so that it can be merged soon.

**Pull request merged**